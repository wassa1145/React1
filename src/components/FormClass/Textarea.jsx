import React, { Component } from 'react';

export class Textarea extends Component {
	render() {
		return <textarea rows="6" className="form__textarea" value={this.props.value} onChange={this.props.change}/>
  }
}
