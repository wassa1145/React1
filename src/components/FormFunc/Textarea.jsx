export const Textarea = (props) => {
	return <textarea rows="6" className="form__textarea" value={props.value} onChange={props.change}/> 
}
