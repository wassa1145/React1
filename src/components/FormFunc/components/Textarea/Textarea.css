.form__textarea {
  width: 100%;
  font-size: 18px;
  padding: 10px;
  font-family: Arial, Helvetica, sans-serif;
  box-sizing: border-box;
  margin-right: 20px;
}
.form__textarea::placeholder {
  font-size: 18px;
  font-family: Arial, Helvetica, sans-serif;
}