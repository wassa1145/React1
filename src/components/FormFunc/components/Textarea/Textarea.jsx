import React from 'react';
import './Textarea.css';

export const Textarea = (props) => {
  return (
    <textarea
      rows="6"
      className="form__textarea"
      placeholder={props.placeholder}
      value={props.message}
      onChange={props.change}
      data-testid="textarea"
    />
  );
};
