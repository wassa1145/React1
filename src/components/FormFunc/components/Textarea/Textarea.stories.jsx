import React from 'react';

import { Textarea } from './Textarea';

export default {
  title: 'MyComponents/Textarea',
  component: Textarea,
};

const Template = (args) => <Textarea {...args} />;

export const Placeholder = Template.bind({});
Placeholder.args = {
  placeholder: 'Введите сообщение',
};

export const Filling = Template.bind({});
Filling.args = {
  placeholder: 'Введите сообщение',
  message: 'Новое сообщение',
};
