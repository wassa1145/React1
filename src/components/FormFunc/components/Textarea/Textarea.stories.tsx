import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Textarea } from './Textarea';

export default {
  title: 'MyComponents/Textarea',
  component: Textarea,
} as ComponentMeta<typeof Textarea>;

const Template: ComponentStory<typeof Textarea> = (args) => (
  <Textarea {...args} />
);

export const Placeholder = Template.bind({});
Placeholder.args = {
  placeholder: 'Введите сообщение',
};

export const Filling = Template.bind({});
Filling.args = {
  placeholder: 'Введите сообщение',
  message: 'Новое сообщение',
};
