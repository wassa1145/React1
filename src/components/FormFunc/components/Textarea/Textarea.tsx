import React, { FC } from 'react';
import TextField from '@mui/material/TextField';
import './Textarea.css';

interface TextareaProps {
  placeholder: string;
  message?: string;
  textFieldRef?: React.RefObject<HTMLInputElement>;
  change: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
}
export const Textarea: FC<TextareaProps> = (props) => {
  return (
    <TextField
      variant="outlined"
      className="form__textarea"
      multiline
      rows={4}
      placeholder={props.placeholder}
      value={props.message}
      onChange={props.change}
      inputRef={props.textFieldRef}
      data-testid="textarea"
    />
  );
};
