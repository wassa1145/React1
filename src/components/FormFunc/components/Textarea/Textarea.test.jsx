import React from 'react';
import { Textarea } from './Textarea';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

describe('Textarea', () => {
  it('render component', () => {
    render(<Textarea />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Textarea />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component', () => {
    render(<Textarea />);
    expect(screen.getByTestId('textarea')).toBeInTheDocument();
  });

  it('input text', async () => {
    render(<Textarea />);
    await userEvent.type(screen.getByTestId('textarea'), 'new message');
    expect(screen.getByTestId('textarea')).toHaveValue('new message');
  });
});
