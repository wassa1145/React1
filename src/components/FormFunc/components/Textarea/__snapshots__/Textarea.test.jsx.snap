// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea render with snapshot 1`] = `
<DocumentFragment>
  <textarea
    class="form__textarea"
    data-testid="textarea"
    rows="6"
  />
</DocumentFragment>
`;
